DataOnion/Auth/TwoFactorAuthService.cs

```diff
@@ -0,0 +1,260 @@
+using Microsoft.Extensions.Logging;
+
+namespace DataOnion.Auth
+{
+    public interface ITwoFactorAuthService
+    {
+        Task<RegisterDidResult> RegisterDidAsync(
+            RegisterDidParams parameters
+        );
+
+        Task<TwoFactorRequest?> FetchRequestAsync(
+            FetchRequestParams parameters
+        );
+
+        Task DeleteRequestsAsync(
+            string userId
+        );
+    }
+
+    public class TwoFactorAuthService : ITwoFactorAuthService
+    {
+        private readonly ITwoFactorRedisContext _redisContext;
+        private readonly ILogger? _logger;
+
+        public TwoFactorAuthService(
+            ITwoFactorRedisContext redisContext,
+            ILogger<TwoFactorAuthService>? logger
```

Comment:
I believe you have to explicitly provide a default `= null`, or else it'll crash at runtime if no logger is available.

Reply:
Added this